client/app/startup/ServerApp.jsx

```diff
@@ -1,17 +1,47 @@
 import React from 'react';
 import { Provider } from 'react-redux';
+import { match, RoutingContext } from 'react-router';
 
 import createStore from '../stores/commentsStore';
-import CommentScreen from '../components/CommentScreen';
+import routes from '../routes/routes';
 
-const App = props => {
+const App = (props, location) => {
   const store = createStore(props);
-  const reactComponent = (
+
+  let routeError;
+  let routeRedirect;
+  let routeProps;
+
+  match({ routes, location }, (_routeError, _routeRedirect, _routeProps) => {
```

Comment:
This run synchronously because we are not setting up the app with dynamic routes.

This does not affect the react_on_rails gem because it does not support dynamic routes.

@robwise FYI -- note for you in the docs.